pgdog/src/admin/show_mirror_stats.rs

```diff
@@ -0,0 +1,221 @@
+                stats
+                    .requests_mirrored
+                    .load(std::sync::atomic::Ordering::Relaxed),
+            ),
+            (
+                "requests_dropped",
+                stats
+                    .requests_dropped
+                    .load(std::sync::atomic::Ordering::Relaxed),
+            ),
+            (
+                "errors_connection",
+                stats
+                    .errors_connection
+                    .load(std::sync::atomic::Ordering::Relaxed),
+            ),
+            (
+                "errors_query",
+                stats
+                    .errors_query
+                    .load(std::sync::atomic::Ordering::Relaxed),
+            ),
+            (
+                "errors_timeout",
+                stats
+                    .errors_timeout
+                    .load(std::sync::atomic::Ordering::Relaxed),
+            ),
+            (
+                "errors_buffer_full",
+                stats
+                    .errors_buffer_full
+                    .load(std::sync::atomic::Ordering::Relaxed),
+            ),
+            (
+                "consecutive_errors",
+                stats
+                    .consecutive_errors
+                    .load(std::sync::atomic::Ordering::Relaxed),
+            ),
+        ];
+
+        for (name, value) in metrics {
+            let mut dr = DataRow::new();
+            dr.add(name).add(value as i64);
+            messages.push(dr.message()?);
+        }
+
+        // Add latency metrics
+        let latency_count = stats
+            .latency_count
+            .load(std::sync::atomic::Ordering::Relaxed);
+        if latency_count > 0 {
+            let latency_sum = stats
+                .latency_sum_ms
+                .load(std::sync::atomic::Ordering::Relaxed);
+            let latency_avg = latency_sum / latency_count;
+            let latency_max = stats
+                .latency_max_ms
+                .load(std::sync::atomic::Ordering::Relaxed);
+
+            let mut dr = DataRow::new();
+            dr.add("latency_avg_ms").add(latency_avg as i64);
+            messages.push(dr.message()?);
+
+            let mut dr = DataRow::new();
+            dr.add("latency_max_ms").add(latency_max as i64);
+            messages.push(dr.message()?);
+        }
+
+        // Add error rate
+        let error_rate = stats.error_rate();
+        let mut dr = DataRow::new();
+        dr.add("error_rate").add(format!("{:.4}", error_rate));
+        messages.push(dr.message()?);
+
+        Ok(messages)
+    }
+}
+
+pub struct ShowMirrorStatsByDatabase;
+
+#[async_trait]
+impl Command for ShowMirrorStatsByDatabase {
+    fn name(&self) -> String {
+        "SHOW MIRROR_STATS_BY_DATABASE".into()
+    }
+
+    fn parse(_: &str) -> Result<Self, Error> {
+        Ok(Self)
+    }
+
+    async fn execute(&self) -> Result<Vec<Message>, Error> {
+        let fields = vec![
+            Field::text("database"),
```

Comment:
The identifier for a "database" is `(database, user)`. We create a `Cluster` for each user/database pair.